### ISSUE TITLE
Different outputs from EQTransformer using obspy.stream and numpy arrayfor the same trace

### ISSUE THREAD

**Issue by Wren-Meta** [seisbench/seisbench#74]
Thanks for the really great work. It is really helpful for me.

Recently, I tried to use the pre_trained EQTransformer model in Pytorch version provided in seisbench. When feeding two different format data (**Stream or numpy array**) for the same trace into the model, I got **different model outputs**. The following steps are included:
[1] import seisbench
     import seisbench.models as sbm
     model = sbm.EQTransformer.from_pretrained("original") 
[2] picks, detection = model.classify(stream)  ## original waveform was converted to obspy stream
     for pick in picks:
         print(pick)
[3] with torch.no_grad():
            detection, p, s = model(torch.tensor(waveform).to(torch.float32))  ## waveform.reshape: (1,3,6000)
            detection = detection.numpy()
            p = p.numpy()
            s = s.numpy()
      pred_p = trace_start_time + np.argmax(p, axis = 1)[0] / 100.0  ## conversion to UTCDateTime
      pred_s = trace_start_time + np.argmax(s, axis = 1)[0] / 100.0           
 
I would like to ask for some suggestions to solve this problem. 

Thanks.

**Comment by yetinam** [comment]
Hey @Wren-Meta ,

thanks for reaching out. The differences between the stream interface and the pytorch interface are expected as they aim at different application scenarios:

- The stream interface is aimed at practitioners. It **incorporates** all pre- and post-processing, i.e., frequency filtering, cutting and reassembling the trace, or normalisation. For example, EQTransformer's original weights add a high-pass filter at 1 Hz.
- The pytorch interface give direct access to the underlying pytorch model **without any** pre-/post-processing. This should usually only be used for training the model or very targeted evaluation.

From your example, including the code you sent via mail, it looks like you are not applying all preprocessing steps when using the pytorch interface. In particular, there seems to be not high-pass filter.

I'd recommend to choose the interface depending on your specific use case. I hope, the information above helps you with this. Feel free to reopen the issue if you have further questions.

**Comment by Wren-Meta** [comment]
Dear yetinam,

Thank you for the information. As you suggested, I will have a try on the Pytorch interface after pre/post-processing .

Best regards,

Wei

On 31. Mar 2022, at 10:57, yetinam ***@***.******@***.***>> wrote:



Hey @Wren-Meta<https://github.com/Wren-Meta> ,

thanks for reaching out. The differences between the stream interface and the pytorch interface are expected as they aim at different application scenarios:

  *   The stream interface is aimed at practitioners. It incorporates all pre- and post-processing, i.e., frequency filtering, cutting and reassembling the trace, or normalisation. For example, EQTransformer's original weights add a high-pass filter at 1 Hz.
  *   The pytorch interface give direct access to the underlying pytorch model without any pre-/post-processing. This should usually only be used for training the model or very targeted evaluation.

From your example, including the code you sent via mail, it looks like you are not applying all preprocessing steps when using the pytorch interface. In particular, there seems to be not high-pass filter.

I'd recommend to choose the interface depending on your specific use case. I hope, the information above helps you with this. Feel free to reopen the issue if you have further questions.
